internal/goldendataset/traces_generator.go

```diff
@@ -86,23 +82,28 @@ func generateLibrarySpansArray(tracingInputs *PICTTracingInputs, spanPairsFile s
 		count = 2
 	}
 	var err error
-	libSpans := make([]*otlptrace.InstrumentationLibrarySpans, count)
+	var libSpans *pdata.InstrumentationLibrarySpans
+	instrumentationLibrarySpansSlice := pdata.NewInstrumentationLibrarySpansSlice()
 	for i := 0; i < count; i++ {
-		libSpans[i], err = generateLibrarySpans(tracingInputs, i, spanPairsFile, random)
+		libSpans, err = generateLibrarySpans(tracingInputs, i, spanPairsFile, random)
+		if err != nil {
+			instrumentationLibrarySpansSlice.Append(*libSpans)
+		}
```

Comment:
This is a bug, fixed in #3377.

Reply:
This bug causes actually to always produce empty data, so that is the reason the previous bug was hidden.